### PR TITLE
Deprecate Atomic[Markable|Stamped]Reference CAS and TM support

### PR DESCRIPTION
**Pull request by fjeremic**
The primate methods supporting CAS and TM in this API were proof of
concept prototypes and no longer exist in the actual Java
implementation. As such this code is slated for deprecation as well as
any code that is dead as a result.

Signed-off-by: Filip Jeremic <fjeremic@ca.ibm.com>

### CODE REVIEW

**Comment by fjeremic** [comment]
@andrewcraik, @gita-omr could I please get a review for x86 and Power?

**Comment by gita-omr** [comment]
Looks fine to me but I would just double check with @ymanton -- was this supposed to be something useful on Power?

**Comment by ymanton** [comment]
No, it can go.

File: runtime/compiler/trj9/x/codegen/J9TreeEvaluator.cpp [hunk 1]
```diff
@@ -11945,965 +11945,42 @@ inlineStringHashCode(
...
-    * not a hinderance, because 64-bit code on IA32 uses register pairs and other
-    * things that are totally different from their 32-bit counterparts.
-    */
-
-   TR_ASSERT(TR::Compiler->target.is64Bit() || node->getSize() <= 4, "64-bit nodes on 32-bit platforms shouldn't use getNodeIs64Bit");
-   return TR::Compiler->target.is64Bit() && node->getSize() > 4;
-   }
-
-static
-TR::Register *intOrLongClobberEvaluate(
```
Comment (andrewcraik) [inline runtime/compiler/trj9/x/codegen/J9TreeEvaluator.cpp]: Did you mean to delete this? It looks more generic than the CS stuff...?

**Comment by fjeremic** [comment]
Jenkins test sanity

**Comment by fjeremic** [comment]
Could we get a committer for this one? Perhaps @andrewcraik or @gita-omr?

**Comment by fjeremic** [comment]
Jenkins test sanity win

**Comment by fjeremic** [comment]
Jenkins test sanity win jdk9

**Comment by fjeremic** [comment]
@AdamBrousseau any idea why the above jdk9 build did not work? The other jdk8 and jdk10 did kick off when I did "Jenkins test sanity win". I would have figured jdk9 would as well.

**Comment by AdamBrousseau** [comment]
win jdk9 PRs are currently disabled because of an openjdk bug. See #1662

**Pull request merged**